chore: add security policy (#1)
opalmedapps/.github: SECURITY.md

# Security Policy

## Reporting a Vulnerability

> 

Reviewer: Maybe we can add a warning here to make it clear right away to not report them on GitHub issues?

Example: https://github.com/nextcloud/server/blob/master/SECURITY.md

```suggestion

> ![WARNING]
> Please do **NOT** report security vulnerabilities through public GitHub issues.

```

If you discover a security vulnerability, we encourage you to report it as soon as possible. Please follow the steps below to report a vulnerability:

1. Do not open an issue on GitHub: Security issues should not be publicly disclosed until they have been addressed. Avoid discussing potential vulnerabilities in public forums such as GitHub issues, forums, or social media.

2. Report the vulnerability to Dr. John Kildea at john.kildea@mcgill.ca. Please include as much information as possible to help us understand the nature and scope of the issue:

>     - A detailed description of the vulnerability.
>     - Steps to reproduce the vulnerability.
>     - Any relevant screenshots or code snippets.
>     - Potential impact of the vulnerability.
>     - Any other relevant information.

Reviewer: ```suggestion
    - A detailed description of the vulnerability
    - Affected component and version
    - Steps to reproduce the vulnerability
    - Any relevant screenshots or code snippets
    - Potential impact of the vulnerability
    - Any other relevant information
```

## Security Updates

Users are encouraged to keep their installations up-to-date to benefit from the latest security fixes.

> 

Reviewer: I don't think this needs to be part of the security policy. Seems out of scope to me.

Author: Just in case some people out there that doesn't like doing updates, that it is not our fault if we fixed the security issue a few version ago and now they have a security problem. If we have this in our security policy, then they can't say that they thought it was just an FYI. Also, since we are dealing with patient data, I thought it would make sense to have it. Showing that we take security seriously.

I know it is common sense, but there are people out there that lack common sense.

It's up to you 😄  We can remove it.

Reviewer: Let's remove it for now, please.

Author: Done

> ## Reporting a bug in a third-party libraries
> 
> For security vulnerabilities found in third-party libraries, please also contact the library's owner or maintainer directly.
> 

Reviewer: Do you mean third-party libraries that we use?

```suggestion
## Reporting a bug in a third-party libraries

For security vulnerabilities found in third-party dependencies that we use in our product, please also contact the dependency's owner or maintainer directly as per their security policy.

```

Author: yep 😄

<!--- Additional option to add

1. Should we have our own PGP encryption when sending information to us?
2. Should we add in the "Reporting a Vulnerability" about them reading a vulnerability on a forum or social media? If so, should we ask them to provide us the details so that we can investigate?

> -->

Reviewer: Let's remove all this for now.

PGP key would be ideal but would require John to set this up. Maybe we can do this once we move to a dedicated email address.

Author: Sound good to me.
